packages/ember-metal/lib/observer_set.js

```diff
@@ -28,28 +25,20 @@ export default class ObserverSet {
       this.added.set(object, keys);
     }
 
-    if (!keys.has(key)) {
-      this.queue.push({ object, key });
-      keys.add(key);
-    }
+    keys.add(key);
   }
 
-  flush() {
-    for (let i = 0; i < this.queue.length; i++) {
-      let { object, key } = this.queue[i];
-
-      if (object.isDestroying || object.isDestroyed) {
-        continue;
+  forEach(cb) {
+    this.added.forEach(function(keys, object) {
+      if (!(object.isDestroying || object.isDestroyed)) {
+        keys.forEach(function(key) {
+          cb(object, key, [object, key]);
```

Comment:
Also I think it is better to just call with `cb(object, key)`. The last argument is specific to `sendEvent`.